core/shared/src/main/scala/spire/math/Polynomial.scala

```diff
@@ -539,6 +539,7 @@ trait PolynomialOverField[@sp(Double) C] extends PolynomialOverRing[C]
     }
   }
 
+  /* TODO: check if we can use directly Euclidean algorithm
   // TODO: why final?
   final override def gcd(x: Polynomial[C], y: Polynomial[C])(implicit ev: Eq[Polynomial[C]]): Polynomial[C] = {
```

Comment:
OK... the trick is that Polynomial.gcd had a special test to return a nice number in case the gcd was a constant.

As the gcd is defined up to a unit, we could keep this postprocessing, maybe extend it to non-constant results. Keeping only the special case feels wrong; let's rediscuss the matter later #641